sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -344,6 +344,59 @@ class Analyzer(
       }
     }
 
+    def buildRightChild4Deduplication (left: LogicalPlan, right: LogicalPlan): LogicalPlan = {
```

Comment:
extra space.  Also I'm not a huge fan of `4`.  How about `dedupRight` with some scala doc.